api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java

```diff
@@ -20,12 +27,67 @@ ListRuntimeResponse toApiListRuntimeResponse(
       LeonardoListRuntimeResponse leonardoListRuntimeResponse);
 
   @Mapping(target = "createdDate", source = "auditInfo.createdDate")
+  @Mapping(target = "toolDockerImage", source = "runtimeImages")
+  @Mapping(target = "configurationType", ignore = true)
+  @Mapping(target = "gceConfig", ignore = true)
+  @Mapping(target = "dataprocConfig", ignore = true)
   Runtime toApiRuntime(LeonardoGetRuntimeResponse runtime);
 
+  default Integer extractIntField(Map<String, Object> map, String field) {
+    if (map.get(field) == null) {
+      return null;
+    }
+
+    return ((Number) map.get(field)).intValue();
+  }
+
+  @AfterMapping
+  default void mapRuntimeConfig(
+      @MappingTarget Runtime runtime, LeonardoGetRuntimeResponse leonardoGetRuntimeResponse) {
+    // There's a lot of unchecked casting happening here but I couldn't find a way around it
+    // This is primarily due to the fact that Swagger generates the `runtimeConfig` object as a
+    // generic Java object.
+    // From manual testing, it seems like it is generated as a LinkedTreeMap.
```

Comment:
You should be able to unmarshal it using Gson. Something like this, but probably simpler: https://github.com/all-of-us/workbench/blob/3fd2793f78387d0bb14f1a804a58d7c93612fc8a/api/src/main/java/org/pmiops/workbench/firecloud/FirecloudTransforms.java#L22